internal/consumer/service.go

```diff
@@ -0,0 +1,82 @@
+package consumer
+
+import (
+	"log"
+	"runtime"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type Item struct {
+	By          string `bson:"by"`
+	Title       string `bson:"title"`
+	Url         string `bson:"url"`
+	Text        string `bson:"text"`
+	ItemType    string `bson:"itemType" json:"type"`
+	Descendants int    `bson:"descendants"`
+	Id          int    `bson:"id"`
+	Score       int    `bson:"score"`
+	Time        int    `bson:"time"`
+	Parent      int    `bson:"parent"`
+	Poll        int    `bson:"poll"`
+	Kids        []int  `bson:"kids"`
+	Parts       []int  `bson:"parts"`
+	Deleted     bool   `bson:"deleted"`
+	Dead        bool   `bson:"dead"`
+}
+
+func Execute(dbRepo DbRepository, dataService DataService) {
+	ids, err := dataService.getTopStories()
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "Consume: "))
+	}
+
+	idChan := make(chan int)
+	itemChan := make(chan Item)
+
+	go populateIdChan(idChan, ids)
+
+	go fanOutIds(idChan, itemChan, dataService)
+
+	for i := range itemChan {
+		err := dbRepo.SaveItem(i)
+		if err != nil {
+			log.Print(errors.Wrap(err, "Consume: "))
+		}
+	}
+}
+
+func fanOutIds(idChan chan int, itemChan chan Item, dataService DataService) {
+	var wg sync.WaitGroup
+	var goRoutines = runtime.NumCPU()
```

Comment:
I understand what you're doing here but this might not be the best thing to do. This is looking at concurrency like multi-threading - I'd pass the worker count through as an arg that can be configured.

Reply:
ok i will update this. thanks